fallback_builtins.h

```diff
@@ -63,4 +63,50 @@ static inline __m512i _mm512_zextsi128_si512(__m128i a) {
 #endif // gcc version 10 test
 
 #endif // __AVX2__
+
+#ifdef ARM_NEON_SLIDEHASH
+
+#define vqsubq_u16_x4_x1(out, a, b) do { \
+    out.val[0] = vqsubq_u16(a.val[0], b); \
+    out.val[1] = vqsubq_u16(a.val[1], b); \
+    out.val[2] = vqsubq_u16(a.val[2], b); \
+    out.val[3] = vqsubq_u16(a.val[3], b); \
+} while (0)
+
+/* Have to check for hard float ABI on GCC/clang, but not 
+ * on MSVC (guess it's assumed to be true? */
```

Comment:
Comment could use some cleaning up.

Reply:
Visual C++ doesn't support 32-bit ARMv8, and only FPU targets it supports for ARMv7 are VFPv3 and VFPv4... It does confusingly have NEON headers, but the compiler itself has no option to enable NEON support or target ARMv8. I filed a bug sometime ago to add support for 32-bit ARMv8 and as such proper command-line support for enabling or disabling NEON explicitly.

Reply:
We do actually have some people from ARM contributing to this repository, but as far I have understood, as ARM instruction set is basically incremental set of features, for me it looks like VFPv3 has less features than VFPv4 (includes FMA+half-precision) and anything after than has more features. ARMv8 only had two optional features, ACLE and NEON.

Reply:
I think you're mostly right, v4 added some extra instructions but I think NEON basically inherited all of those instructions but was an actual SIMD ISA in execution (looks like wikipedia says VFPv4 operated sequentially).  I don't _think_ NEON is an optional component of ARMv8, but ACLE definitely is (which is why people were sort of annoyed that the pi4 didn't implement those instructions).  

https://developer.arm.com/documentation/den0024/a/AArch64-Floating-point-and-NEON

lol, of course, confusingly, "specialized" devices can be without it.  They also had a fun quirk where they somehow murdered performance on the TBL instruction (the aarch64/aarch32 equivalent of the shuffle instructions).  I'm not sure if every Cortex SIP suffers from this, but A72 definitely did.

Reply:
32-bit only ARMv8 processors don't always have NEON... There is also another vector extension (Helium) used by some later ARMv8.1 processors that don't support NEON.

Reply:
Yes, but from what I understand helium was made for lower powered CPUs bearing the -M moniker.

Reply:
True.... And that leaves us ARMv8-A and ARMv8-M when it comes to possible variations. With ARMv7 and earlier, there was many more variations, not forgetting VFPv3, VFPv4 and VFPv5 for the co-processor.